WooCommerce/Classes/ViewModels/Order Details/OrderDetailsPaymentAlerts.swift

```diff
@@ -6,71 +6,79 @@ import WordPressUI
 /// presented to provide user-facing feedback about the progress
 /// of the payment collection process
 final class OrderDetailsPaymentAlerts {
-    private var modalController: CardPresentPaymentsModalViewController?
+    private weak var presentingController: UIViewController?
+
+    // Storing this as a weak variable means that iOS should automatically set this to nil
+    // when the VC is dismissed, unless there is a retain cycle somewhere else.
+    private weak var _modalController: CardPresentPaymentsModalViewController?
+    private var modalController: CardPresentPaymentsModalViewController {
```

Comment:
Right, you want to make the reference weak... sorry!

Reply:
I'd have to double check, but if my memory is correct, you can use `weak lazy var`. However, I understand that `lazy` will only run the initializer block the first time the variable is accessed. If it's set to `nil` afterwards it will stay that way.